Updates to move busulfan and PEG-Asp to new framework

jasmineirx opened the pull request:
- Made Peg Asparaginase and Busulfan vignettes work with new framework
- Added text to make it clearer which vignettes use which techniques
- Added a function to allow for weight-based dosing nomograms for initial doses. We can easily extend for BSA-based, etc.
- Added an example for how to use a custom initial dose function (yay functional programming)

Still some clean-up to be done to get this merged but I thought this was a reasonable amount to review to start.

mccarthy-m-g commented:
Do you mind if I commit some revisions to the vignette?

jasmineirx commented:
sure! i'm working on getting some of the unit tests working right now. They're not failing due to change I made here. it's due to other issues in the base feature branch.

mccarthy-m-g commented:
I made some small changes to prose and code for better flow. I also added a few `FIXME` comments to the vignette that you might want to address.

Regarding the internal data set `FIXME` comment: If you haven't added a data set to a package before, let me know and I'd be happy to do it!

jasmineirx commented:
I'm a bit hesitant to add a data set to a package for only one vignette. wdyt?

mccarthy-m-g commented:
That's one of the most common purposes of adding data sets to a package 😄 I do think it would make the vignette a lot easier to read---the wrangling code there distracts from the purpose of what you're demonstrating.

It takes little effort, no maintenance, and just a small bit of documentation. We basically just need to move the wrangling code you've already done to a script in a `data-raw` directory, which will call `usethis::use_data()` on the final data set. Then just add an R script with roxygen documentation for the data and a pkgdown reference.

JordanBrooks33 reviewed (approved):
lgtm. Regarding moving the dataset code/including a pre-drawn dataset, I think it sounds reasonable if minimal. Though to be honest, for the intended audience of the package, I don't the data being pulled within the vignettes as is will be a hang up and may actually be more readily transparent/easy to follow.

roninsightrx commented on R/dosing_nomograms.R:
```diff
@@ -0,0 +1,23 @@
+    "specify `create_initial_regimen_design(mgkg = ...)`"
+    )
+  }
+  if (is.null(covariates$WT$value)) {
```
should we use covariate mapping here? The user does create a covariate map. Not sure if it makes it easier / more complex.

jasmineirx replied:
we could make it customizable which covariate is used for weight, but it would have to be specified in the `create_initial_regimen_design` call and not in cov_mapping, since cov_mapping is model-to-data mapping and does not necessary contain info about weight.

we could imagining something like this:

```r
  init_reg <- create_initial_regimen_design(
    method = weight_based_starting_dose,
    mgkg = 15,
    weight_cov = "WT",
    regimen = list(
      interval = 12,
      t_inf = 1.5,
      n = 10,
      type = "infusion"
    )
  )
```

This would allow the user to also do the following to calculate it based on something like IBW:
```r
  init_reg <- create_initial_regimen_design(
    method = weight_based_starting_dose,
    mgkg = 15,
    weight_cov = "IBW",
    regimen = list(
      interval = 12,
      t_inf = 1.5,
      n = 10,
      type = "infusion"
    )
  )
```

roninsightrx replied:
Yeah like this, more generic. But perhaps `weight_cov` doesn't need to be an argument to `create_initial_regimen_design`, can we just depend on passing via `...`? OTOH weight is often useful for initial dose, so I don't mind adding it as an argument.

jasmineirx replied:
@roninsightrx I have updated this function to be more generic, it will be passed to `create_initial_regimen_design` with `...` args, but it will be required if this method is called, for obvious reasons.

roninsightrx reviewed (approved):
lgtm, one question

jasmineirx commented:
Hey, changed to predefining the data using `constructive::construct` and removed the NHANES Suggests, since this seems to balance simplicity and readability well.

Tests are still failing but I'll look into it tomorrow.

mccarthy-m-g reviewed (commented):
Nice! I think that's easier to follow now.

JordanBrooks33 reviewed (approved):
lgtm
